tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java

```diff
@@ -77,31 +81,71 @@ public UpdateTicketHandler() {
 
     protected UpdateTicketHandler(TicketService ticketService, ResourceService resourceService, DoiClient doiClient,
                                   Environment environment) {
-        super(UpdateTicketRequest.class, environment);
+        super(TicketRequest.class, environment);
         this.ticketService = ticketService;
         this.resourceService = resourceService;
         this.doiClient = doiClient;
     }
 
     @Override
-    protected void validateRequest(UpdateTicketRequest updateTicketRequest, RequestInfo requestInfo, Context context)
+    protected void validateRequest(TicketRequest ticketRequest, RequestInfo requestInfo, Context context)
         throws ApiGatewayException {
         //Do nothing
     }
 
     @Override
-    protected Void processInput(UpdateTicketRequest input, RequestInfo requestInfo, Context context)
+    protected Void processInput(TicketRequest input, RequestInfo requestInfo, Context context)
         throws ApiGatewayException {
         var requestUtils = RequestUtils.fromRequestInfo(requestInfo);
         var ticket = fetchTicket(requestUtils);
-        if (hasEffectiveChanges(ticket, input)) {
-            updateTicket(input, requestUtils, ticket);
+
+        switch (input) {
+            case UpdateTicketRequest request -> handleUpdateTicketRequest(ticket, request, requestUtils);
+            case UpdateTicketOwnershipRequest request -> updateOwnership(ticket, request, requestUtils);
+            default -> {
+                // NO OP
+            }
         }
+
         return null;
     }
 
+    private void updateOwnership(TicketEntry ticket, UpdateTicketOwnershipRequest request,
+                                 RequestUtils requestUtils) throws ForbiddenException, NotFoundException {
+        var resource = getResource(ticket.getResourceIdentifier());
+        if (userInstitutionIsTicketOwner(ticket, requestUtils)
+            && newOwnerAffiliationIsOneOfCuratingInstitutions(request, resource)
+            && userIsAuthorized(requestUtils, ticket)) {
```

Comment:
This conditional is ugly, alternative is to extract to method with 4 variables as input.